test/client/fixtures/sandbox/node/live-node-list-factory-test.js

```diff
@@ -260,63 +210,111 @@ module('getElementsByTagName', function () {
-                    return refreshNodeListCount === 1;
-                })()
-            });
+            assertions.push([refreshNodeListCount, 1, 'tracking tagName']);
 
             document.body.replaceChild(input1, textarea1);
 
-            assertions.push({
-                name:  'replaceChild for tracking and non-tracking nodes',
-                value: (function () {
-                    elements[0];
+            elements[0];
 
-                    return refreshNodeListCount === 2;
-                })()
-            });
+            assertions.push([refreshNodeListCount, 2, 'replaceChild for tracking and non-tracking nodes']);
+
+            setProperty(testDiv, 'innerHTML', '<div><textarea></textarea></div>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 3, "access after set element's innerHTML"]);
+
+            setProperty(testDiv.firstChild, 'innerText', 'text');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 4, "access after set element's innerText"]);
+
+            setProperty(testDiv.firstChild, 'outerHTML', '<div></div>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 4, "access after set element's outerHTML without textarea"]);
+
+            setProperty(testDiv.firstChild, 'outerHTML', '<textarea></textarea>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 5, "access after set element's outerHTML with textarea"]);
+
+            var nativeDiv      = nativeMethods.createElement.call(document, 'div');
+            var nativeTextArea = nativeMethods.createElement.call(document, 'textarea');
+
+            nativeMethods.appendChild.call(nativeDiv, nativeTextArea);
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 5, 'access when no changes']);
+
+            testDiv.appendChild(nativeDiv);
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 6, 'access after element was added with another element']);
+
+            testDiv.removeChild(nativeDiv);
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 7, 'access after element was removed with another element']);
+
+            testDiv.insertAdjacentHTML('beforebegin', '<textarea></textarea>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 8, 'access after element was added before begin text div']);
+
+            testDiv.insertAdjacentHTML('afterbegin', '<textarea></textarea>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 9, 'access after element was added after begin text div']);
+
+            testDiv.insertAdjacentHTML('beforeend', '<textarea></textarea>');
+
+            elements[0];
+
+            assertions.push([refreshNodeListCount, 10, 'access after element was added before end text div']);
+
+            testDiv.insertAdjacentHTML('afterend', '<textarea></textarea>');
```

Comment:
Add test for `documentFragment`

Reply:
Test for `documentFragment` does not work, need to fix [this](https://github.com/DevExpress/testcafe-hammerhead/issues/1334) issue.

Reply:
Ok